src/Microsoft.AspNet.Mvc.Core/ModelBinding/DictionaryModelBinder.cs

```diff
@@ -80,15 +84,34 @@ public class DictionaryModelBinder<TKey, TValue> : CollectionModelBinder<KeyValu
         }
 
         /// <inheritdoc />
-        protected override object GetModel(IEnumerable<KeyValuePair<TKey, TValue>> newCollection)
+        protected override object GetModel(Type targetType, IEnumerable<KeyValuePair<TKey, TValue>> collection)
         {
-            return newCollection?.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+            if (collection == null)
+            {
+                return null;
+            }
+
+            if (targetType.IsAssignableFrom(typeof(Dictionary<TKey, TValue>)))
```

Comment:
A `List<KeyValuePair<TKey, TValue>>`. Are you asking for a comment stating that it's never already a dictionary? Note the parameter is used only as it's declared -- `IEnumerable<KeyValuePair<TKey, TValue>>`.